### DISCUSSION TITLE
What's the drop rate of Decapitation?

### DISCUSSION THREAD

**Discussion by Uboaaaaa** [Geolykt/EnchantmentsPlus#59] [Q&A] [answered]
Hi, I was thinking of enabling decapitation on my server but was concerned about how effective it is in dropping an skull. Do you have any idea on the exact value of the rate?

I also think it will be very helpful if we can know some of the behind-the-scene data of the enchantments, such as the experience multiplier of _leveling_, nugget drop rate of _gold rush_, _siphon_'s healing ratio etc. It helps a lot when configuring the power multiplier.

**Comment by Geolykt** [comment, marked as answer]
For decapitation the drop formula is
`ThreadLocalRandom.current().nextInt(Math.max((int) Math.round(BASE_MOB_DROP_CHANCE / (level * power)), 1)) == 0`
Since BASE_MOB_DROP_CHANCE is an unconfigurable constant defined to 150 if you were to lower the power of the enchantment by half, the drop chance would be cut in half.
For the power of 1.0 (default) the random function has to choose in between 150 values for level 1, 75 for level 2, 32 for level 3 and so on.
For level two it is 300 values for level 1 and so on.

For siphon the healing rate is `0.17 * level * power` of the damage that was performed, depending on the configuration that can be the final damage but also the original damage.

The multiplier for level is `* (1.3 + (level * power * .5))`

And the drop rate of Gold rush is `level * power * 3`, however note that the enchantment is more granular because it is in steps of 1%

**Comment by Uboaaaaa** [comment]
Thanks for sharing these details! These are absolutely helpful and should be documented somewhere some day.
